pkg/globalmanager/controllers/federatedlearning/federatedlearningjob.go

```diff
@@ -474,39 +570,22 @@ func (c *Controller) createPod(job *sednav1.FederatedLearningJob) (active int32,
 
 	// deliver pod for training worker
 	for i, trainingWorker := range job.Spec.TrainingWorkers {
-		// get dataseturl through parsing crd of dataset
+		// Configure training worker's mounts and envs
+		var workerParam runtime.WorkerParam
+
+		c.addWorkerMount(&workerParam, model.Spec.URL, "MODEL_URL", modelSecret, false)
+
+		c.addWorkerMount(&workerParam, pretrainedModel.Spec.URL, "PRETRAINED_MODEL_URL",
+			pretrainedModelSecret, true)
+
 		datasetName := trainingWorker.Dataset.Name
-		dataset, err := c.client.Datasets(job.Namespace).Get(ctx, datasetName, metav1.GetOptions{})
+		dataset, datasetSecret, err := c.getDataset(ctx, job.Namespace, datasetName)
 		if err != nil {
-			return active, fmt.Errorf("failed to get dataset %s: %w",
-				datasetName, err)
-		}
-
-		secretName := dataset.Spec.CredentialName
-		var datasetSecret *v1.Secret
-		if secretName != "" {
-			datasetSecret, _ = c.kubeClient.CoreV1().Secrets(job.Namespace).Get(context.TODO(), secretName, metav1.GetOptions{})
+			return active, err
 		}
 
-		// Configure training worker's mounts and envs
-		var workerParam runtime.WorkerParam
-		workerParam.Mounts = append(workerParam.Mounts,
-			runtime.WorkerMount{
-				URL: &runtime.MountURL{
-					URL:    model.Spec.URL,
-					Secret: modelSecret,
-				},
-				EnvName: "MODEL_URL",
-			},
-
-			runtime.WorkerMount{
-				URL: &runtime.MountURL{
-					URL:    dataset.Spec.URL,
-					Secret: datasetSecret,
-				},
-				EnvName: "TRAIN_DATASET_URL",
-			},
-		)
+		c.addWorkerMount(&workerParam, dataset.Spec.URL, "TRAIN_DATASET_URL",
+			datasetSecret, false)
```

Comment:
Why is `downloadByInitializer` false for dataset here?

Reply:
change to `true`